assets/js/payment-methods-demo/payment-methods/stripe/payment-method.js

```diff
@@ -0,0 +1,317 @@
+	stripe,
+} ) => {
+	const [ savedPaymentMethodToken, setSavedPaymentMethodToken ] = useState(
+		null
+	);
+	const [ source, setSource ] = useState( null );
+	const customerPaymentMethods = getCustomerPaymentMethods(
+		'cc',
+		PAYMENT_METHOD_NAME
+	);
+	const onStripeError = ( event ) => {
+		const type = event.error.type;
+		const code = event.error.code || '';
+		let message = getErrorMessageForTypeAndCode( type, code );
+		message = message || event.error.message;
+		paymentStatus.setPaymentStatus().error( message );
+		setSource( null );
+
+		// @todo we'll want to do inline invalidation errors for any element
+		// inputs
+	};
+	const updateCardBrand = ( brand ) => {
+		// @todo this updates the card brand logo with non-inline CC forms
+		// (see function in stripe.js)
+	};
+	const onChange = ( paymentEvent ) => {
+		if ( paymentEvent.elementType === 'cardNumber' ) {
+			updateCardBrand( paymentEvent.brand );
+		}
+		if ( paymentEvent.error ) {
+			onStripeError( paymentEvent );
+		} else {
+			setSource( null );
+		}
+	};
+	const renderedCardElement = getStripeServerData().inline_cc_form ? (
+		<CardElement
+			style={ elementStyles }
+			onChange={ onChange }
+			hidePostalCode={ true }
+		/>
+	) : (
+		<CardElements
+			style={ elementStyles }
+			classes={ elementClasses }
+			onChange={ onChange }
+		/>
+	);
+	useEffect( () => {}, [] );
+
+	const createSource = async ( stripeBilling ) => {
+		return await stripe.createSource( stripeBilling );
+	};
+	const onSubmit = () => {
+		const { billingData } = billing;
+		const stripeBilling = {
+			address: {
+				line1: billingData.address_1,
+				line2: billingData.address_2,
+				city: billingData.city,
+				state: billingData.state,
+				postal_code: billingData.postcode,
+				country: billingData.country,
+			},
+		};
+		if ( billingData.phone ) {
+			stripeBilling.phone = billingData.phone;
+		}
+		if ( billingData.email ) {
+			stripeBilling.email = billingData.email;
+		}
+		if ( billingData.first_name || billingData.last_name ) {
+			stripeBilling.name = `${ billingData.first_name } ${ billingData.last_name }`;
+		}
+
+		const response = createSource( stripeBilling );
+		if ( response.error ) {
+			// @todo set payment error
+			// return error object including any validation errors if present?
+		}
+		paymentStatus.setPaymentStatus().success( billingData, {
+			sourceId: response.source.id,
+		} );
+
+		// this should hook into checkout processing? Basically when the checkout
+		// processing button is clicked
+		// if saved source (card) is selected by user, then there should be no
+		// processing/validation by stripe.
+		// as a part of the submission, go to `createSource`
+
+		// if submission completes okay, then set payment status to complete
+		// and that's where billing data and payment data are passed through.
+	};
+	const onComplete = () => {
+		// @todo this is invoked on checkout complete success
+		// in this place just set the currentPaymentStatus to completed.
```

Comment:
and redirect somewhere?

Reply:
Checkout handles the redirect in this case. `onComplete` is hooked into the `onCheckoutCompleteSuccess` event checkout fires as a final set of things to fire before checkout redirects.  It's quite possible for this particular payment method I don't _need_ to do anything here and can remove this (since checkout won't care if the payment status is complete or not).